code/modules/overmap/ftl_shunt/core.dm

```diff
@@ -0,0 +1,551 @@
+
+/obj/machinery/ftl_shunt/core/proc/handle_spacefloat(var/mob/living/carbon/human/H)
+	if(!H.check_space_footing())
+		 //Flip a coin ...
+		to_chat(H, SPAN_WARNING("Being untethered from a ship entering FTL is a bad idea, but you roll the dice..."))
+		if(prob(50))
+			to_chat(H, SPAN_NOTICE("and win, surviving the energy dancing over your body. Not unharmed, however."))
+			H.apply_damage(300, IRRADIATE, damage_flags = DAM_DISPERSED)
+			return
+		else
+			to_chat(H, SPAN_DANGER("and lose, being ripped apart in a nanosecond by energies beyond comprehension."))
+			if(H.isSynthetic())
+				new /obj/item/remains/robot(get_turf(H))
+			else
+				new /obj/item/remains/human(get_turf(H))
+			qdel(H)
+
+/obj/machinery/ftl_shunt/core/proc/do_sabotage()
+	var/announcetxt
+	var/uhoh = FALSE
+
+	switch(sabotaged)
+		if(SHUNT_SABOTAGE_MINOR)
+			announcetxt = shunt_sabotage_text_minor
+			for(var/mob/living/carbon/human/H in view(7))
+				to_chat(H, SPAN_DANGER("[src] emits a flash of incredibly bright, searing light!"))
+				H.flash_eyes(FLASH_PROTECTION_NONE)
+			empulse(src, 8, 10)
+
+		if(SHUNT_SABOTAGE_MAJOR)
+			announcetxt = shunt_sabotage_text_major
+
+			for(var/mob/living/carbon/human/H in view(7)) //Effect One: scary text.
+				to_chat(H, SPAN_DANGER("[src] hisses and sparks, before coolant lines burst and spew superheated coolant!"))
+
+			explosion(get_turf(src),-1,-1,8,10) //Effect Two: blow the windows out.
+
+			for(var/obj/machinery/power/apc/A in SSmachines.machinery) //Effect Three: shut down power across the ship.
+				if(!(A.z in ftl_computer.linked.map_z))
+					continue
+				A.energy_fail(rand(60,80))
+
+		if(SHUNT_SABOTAGE_CRITICAL)
+			announcetxt = shunt_sabotage_text_critical
+
+			uhoh = TRUE //let the fuckening begin!
+
+			for(var/obj/machinery/power/apc/A in SSmachines.machinery) //Effect One: shut down power across the ship.
+				if(!(A.z in ftl_computer.linked.map_z))
+					continue
+				A.energy_fail(rand(100,120))
+
+			for(var/mob/living/carbon/human/H in view(7)) //scary text if you're in view, because you're fucked now boy.
+				to_chat(H, SPAN_DANGER("The light around [src] warps before it emits a flash of incredibly bright, searing light!"))
+				H.flash_eyes(FLASH_PROTECTION_NONE)
+
+
+	ftl_announcement.Announce(announcetxt, "FTL Shunt Management System", new_sound = sound('sound/misc/ftlsiren.ogg'))
+
+	if(uhoh)
+		new /obj/singularity/(get_turf(src))
+
+//Returns status to ftl computer.
+/obj/machinery/ftl_shunt/core/proc/get_status()
+	if(stat & (BROKEN|NOPOWER))
+		return FTL_STATUS_OFFLINE
+	if(cooldown)
+		return FTL_STATUS_COOLDOWN
+	else
+		return FTL_STATUS_GOOD
+
+/obj/machinery/ftl_shunt/core/proc/get_fuel(var/list/input)
+	var/total_fuel_joules
+
+	for(var/obj/machinery/ftl_shunt/fuel_port/F in input)
+		total_fuel_joules += F.get_fuel_joules(FALSE)
+
+	return total_fuel_joules
+
+/obj/machinery/ftl_shunt/core/proc/get_fuel_maximum(var/list/input)
+	var/max_fuel_joules = 0
```

Comment:
Not blocking, but `. = 0` and just using `.` in the loop would remove the need for an explicit return and avoid a var allocation.